### PR TITLE
Convert to TypeScript

### PR DESCRIPTION
**Pull request by Turbo87**
The `any` type is still used quite a lot, but this is a first step that allows us to use better type interfaces as discussed via Slack.

### CODE REVIEW

**Comment by Turbo87** [comment]
@hzoo any comments?

File: package.json [hunk 1]
```diff
@@ -41,17 +41,14 @@
...
-    "eslint-config-babel": "^6.0.0",
-    "eslint-plugin-flowtype": "^2.30.0",
+    "@types/jest": "^19.2.3",
+    "@types/node": "^7.0.22",
     "fs-extra": "^2.0.0",
     "jest": "^20.0.4",
-    "rimraf": "^2.5.4"
+    "rimraf": "^2.5.4",
+    "ts-jest": "^20.0.4",
+    "tslint": "^5.3.2",
```
Comment (hzoo) [inline package.json]: I was going to say you can try eslint + typescript parser + prettier now instead but can do that later

File: src/ConfigurationError.ts [hunk 2]
```diff
@@ -4,10 +4,13 @@
 //
 // Gotta do this the old-fashioned way. :p
 //
-export default function ConfigurationError(message) {
-  this.name = "ConfigurationError";
-  this.message = message;
-  this.stack = (new Error()).stack;
+export default class ConfigurationError {
```
Comment (hzoo) [inline src/ConfigurationError.ts]: didn't really like extending built-ins ever but ok 😄

File: package.json (hunk 1 above)
Comment (Turbo87) [inline package.json]: yeah, I'd like to postpone that for now and focus on the other commits I still have in the pipeline. I'll keep it in mind for later though.

File: src/ConfigurationError.ts (hunk 2 above)
Comment (Turbo87) [inline src/ConfigurationError.ts]: fixed

**Comment by Turbo87** [comment]
since Travis is still 🍏 after addressing the comments I'll merge this and prepare the next PR :)

**Pull request merged**